src/components/color_generator/index.js

```diff
@@ -6,49 +6,76 @@ class ColorGenerator extends Component {
     this.state = {
       id: 0,
       palette_name: '',
-      color1: {hex: '', lockType: 'lock-open'},
-      color2: {hex: '', lockType: 'lock-open'},
-      color3: {hex: '', lockType: 'lock-open'},
-      color4: {hex: '', lockType: 'lock-open'},
-      color5: {hex: '', lockType: 'lock-open'}
+      colors: [{}, {}, {}, {}, {}]
     }
   }
 
-  handleName = (e) => {
-    const { name, value } = e.target;
+  componentDidMount() {
+    this.generateColors()
```

Comment:
Do we want this to be invoked on componentDidMount even when the user is loading an existing palette?

Reply:
I didn't think of that, that's a good point.
- We could maybe refactor it to where: if Hex <string> is blank, then run generate colors.